### DISCUSSION TITLE
Is it possible to have dynamic keys with VRL?

### DISCUSSION THREAD

**Discussion by camilisette** [vectordotdev/vector#18504] [Q&A] [answered]
For example I have this code where I would like the value of the variable `field_name` to be the key name to be used to tag my logs.
```
field_name = "something"

.fields[field_name] = "this is a test"
```
This would result to a log that looks like this
```json
{
    "fields": {
          "something": "this is a test"
     }
    ......
}
```
Thanks in advance!

**Comment by jszwedko** [comment, marked as answer]
Hey! Kinda. See https://vector.dev/docs/reference/vrl/functions/#set

**Reply by camilisette** [reply]
can't believe I didn't notice this. saw that there's one for `get` too. thanks!
